linkerd/io/src/lib.rs

```diff
@@ -76,3 +76,10 @@ impl PeerAddr for tokio::io::DuplexStream {
         Ok(([0, 0, 0, 0], 0).into())
     }
 }
+
+impl<T: PeerAddr> PeerAddr for hyper_util::rt::tokio::TokioIo<T> {
+    #[inline]
+    fn peer_addr(&self) -> Result<SocketAddr> {
+        self.inner().peer_addr()
+    }
+}
```

Comment:
```suggestion
```

Reply:
that's a great catch, thank you. indeed, this is no longer load-bearing! i have removed the commit adding this (8875264) from this branch, along with the associated dependency ([below](https://github.com/linkerd/linkerd2-proxy/pull/3504#discussion_r1999690176)).